string agent_name
string agent_type
